node/src/accountant/payment_adjuster/mod.rs

```diff
@@ -0,0 +1,2340 @@
+        let disqualification_limit_2 =
+            disqualification_arbiter.calculate_disqualification_edge(&account_2);
+        // This is exactly the amount which will provoke an error
+        let cw_service_fee_balance_minor = disqualification_limit_2 - 1;
+        let qualified_payables = vec![account_1, account_2, account_3];
+        let analyzed_payables = convert_collection(qualified_payables);
+        let agent = BlockchainAgentMock::default()
+            .service_fee_balance_minor_result(cw_service_fee_balance_minor);
+        let adjustment_setup = PreparedAdjustment {
+            agent: Box::new(agent),
+            adjustment_analysis: AdjustmentAnalysis::new(
+                Adjustment::TransactionFeeInPriority {
+                    affordable_transaction_count: 2,
+                },
+                analyzed_payables,
+            ),
+            response_skeleton_opt: None,
+        };
+
+        let result = subject.adjust_payments(adjustment_setup, now);
+
+        let err = match result {
+            Ok(_) => panic!("expected an error but got Ok()"),
+            Err(e) => e,
+        };
+        assert_eq!(
+            err,
+            PaymentAdjusterError::LateNotEnoughFeeForSingleTransaction {
+                original_number_of_accounts: 3,
+                number_of_accounts: 2,
+                original_service_fee_required_total_minor: balance_1 + balance_2 + balance_3,
+                cw_service_fee_balance_minor
+            }
+        );
+        TestLogHandler::new().assert_logs_contain_in_order(vec![
+            &format!(
+                "WARN: {test_name}: Total of 411,000,000,000,000,000,000 wei in MASQ was \
+            ordered while the consuming wallet held only 70,999,999,999,999,999,999 wei of MASQ \
+            token. Adjustment of their count or balances is required."
+            ),
+            &format!(
+                "INFO: {test_name}: Please be aware that abandoning your debts is going to \
+            result in delinquency bans. In order to consume services without limitations, you \
+            will need to place more funds into your consuming wallet.",
+            ),
+            &format!(
+                "ERROR: {test_name}: {}",
+                LATER_DETECTED_SERVICE_FEE_SEVERE_SCARCITY
+            ),
+        ]);
+    }
+
+    struct TestConfigForServiceFeeBalances {
+        // Either gwei or wei
+        account_balances: Either<Vec<u64>, Vec<u128>>,
+        cw_balance_minor: u128,
+    }
+
+    struct TestConfigForTransactionFees {
+        agreed_transaction_fee_per_computed_unit_major: u64,
+        number_of_accounts: usize,
+        estimated_transaction_fee_units_per_transaction: u64,
+        cw_transaction_fee_balance_major: u64,
+    }
+
+    fn make_input_for_initial_check_tests(
+        service_fee_balances_config_opt: Option<TestConfigForServiceFeeBalances>,
+        transaction_fee_config_opt: Option<TestConfigForTransactionFees>,
```

Comment:
If we can just call them:

`service_fee_config_opt` and `tx_fee_config_opt`

It'll be symmetric.